### PR TITLE
Fixing outdated `cli.py` documentation

### PR DESCRIPTION
**Pull request by noklam**
Signed-off-by: noklam <nok.lam.chan@quantumblack.com>
Mentioned  in https://github.com/kedro-org/kedro/issues/1474
## Description

`cli.py` doc is outdated.

## Development notes

* update `cli.py` doc and move the snippets to the more common command line reference page.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1481"></a>

### CODE REVIEW

File: docs/source/development/commands_reference.md [hunk 1]
```diff
@@ -137,15 +137,119 @@ kedro new
 kedro docs
 ```
 
-## Project-specific Kedro commands
+## Customize Project-specific Kedro commands
```
Comment (datajoely) [inline docs/source/development/commands_reference.md:140]:
```suggestion
## Customize or Override Project-specific Kedro commands
```

Comment (datajoely) [inline docs/source/development/commands_reference.md:140]:
```suggestion
## Customize or Override Project-specific Kedro commands
```

File: docs/source/development/commands_reference.md [hunk 2]
```diff
@@ -137,15 +137,119 @@ kedro new
...
+    to_outputs,
+    load_version,
+    pipeline,
+    config,
+    params,
+):
+    """Run the pipeline."""
+    """Run the pipeline."""
+
+    == ADD YOUR CUSTOM RUN COMMAND CODE HERE ==
```
Comment (datajoely) [inline docs/source/development/commands_reference.md:249]: I think we should include valid python syntax here, perhaps as a comment? Should we also do the rest of it (`session.run` etc) so they can tweak a working version?

**Review by datajoely — CHANGES_REQUESTED** [review]

File: docs/source/development/commands_reference.md (hunk 2 above)
Comment (noklam) [reply docs/source/development/commands_reference.md:249]: We can keep a separate file in the repo, but it's still gonna be maintain manually as it is a subset of https://github.com/kedro-org/kedro/blob/main/kedro/framework/cli/project.py.

File: docs/source/development/commands_reference.md [hunk 3]
```diff
@@ -137,15 +137,119 @@ kedro new
...
 Kedro's command line interface (CLI) allows you to associate a set of commands and dependencies with a target, which you can then execute from inside the project directory.
 
-The commands a project supports are specified on the framework side. If you want to customise any of the Kedro commands you can do this either by adding a file called `cli.py` or by injecting commands into it via the [`plugin` framework](../extend_kedro/plugins.md). Find the template for the `cli.py` file [here](../extend_kedro/common_use_cases.md#use-case-3-how-to-add-or-modify-cli-commands).
+The commands a project supports are specified on the framework side. If you want to customise any of the Kedro commands you can do this either by adding a file called `cli.py` or by injecting commands into it via the [`plugin` framework](../extend_kedro/plugins.md). Find the template for the `cli.py` file below.
+
+<details>
+<summary><b>Click to expand</b></summary>
+
+```
+"""Command line tools for manipulating a Kedro project.
```
[the remark this answers is not in the document]
Comment (noklam) [reply docs/source/development/commands_reference.md:154]: There is no good place to place this file currently. I think we can group this into https://github.com/kedro-org/kedro/issues/1474, once we have template for these optional files. Either `kedro template xxx` or just a comprehensive starter project which include all optional files like `cli.py`, `logging.yml` etc.

Will just keep it in the doc for now.

File: docs/source/development/commands_reference.md [hunk 4]
```diff
@@ -137,15 +137,139 @@ kedro new
 kedro docs
 ```
 
-## Project-specific Kedro commands
+## Customize or Override Project-specific Kedro commands
```
Comment (SajidAlamQB) [inline docs/source/development/commands_reference.md:140]:
```suggestion
## Customise or Override Project-specific Kedro commands
```

**Review by SajidAlamQB — APPROVED** [review]
LGTM 💥

**Review by datajoely — APPROVED** [review]
🔥

**Pull request merged**